File: web3-libs/mod3/playground/aula1/sign_message.py
from eth_account.messages import encode_defunct
from web3 import Account


# Function to sign a message
def sign_message():
    message = input("Enter the message to sign: ")
    msg = encode_defunct(text=message)
    # Create Account
    account = Account().create()
    print("Address:", account.address)

    # Sign the message
    signature = Account().sign_message(msg, account.key)
    print("Signature:", signature.signature.hex())

    # Verify the signature
    message = "outra messagem"
    msg2 = encode_defunct(text=message)
    verify = Account().recover_message(msg, signature=signature.signature)

    if verify == account.address:
        print("✅ Verification: Success")
    else:
        print("🚨 Verification: Failed")


sign_message()

File: web3-libs/mod4/playground/aula1/signMessage.js
import { ethers, Signature, verifyMessage } from "ethers";
import readline from "readline";

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

// Function to sign a message
const signMessage = async () => {
  rl.question("Enter the message to sign: ", async (message) => {
    // Create Account
    const signer = ethers.Wallet.createRandom();
    console.log("Account:", signer.address);

    // Sign the message
    const rawSig = await signer.signMessage(message);
    console.log("Signature (raw):", rawSig);

    // Split the signature
    const sig = Signature.from(rawSig);
    console.table({ v: sig.v, r: sig.r, s: sig.s });

    const result = verifyMessage(message, rawSig);

    console.log(
      "Verification:",
      result === signer.address ? "Success" : "Failed"
    );

    rl.close();
  });
};

// Call the function to sign a message
signMessage();
